### PR TITLE
cpu/saml1x: fix RTT issue in init process

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description

During the review of #14416, I noticed a bug regarding the RTT module.
This module will get stuck during its init process if the RTC was running from a previous boot.
The RTC module is only reset at power on reset so when you flash tests/periph_rtc then tests/periph_rtt, the RTC module will still be running. This PR fixes the clock management of the RTC module to prevent disabling this clock if the RTC was running as the IP doesn't like this. I also add a workaround to early disable the RTC if it was running only if RTC or RTT is in used in the code.

### Testing procedure
Test the following with master
make BOARD=saml11-xpro -C tests/periph_rtc flash
make BOARD=saml11-xpro -C tests/periph_rtt flash

the RTC output should be good but the RTT should be stuck in the init.
There is no way to recover unless you unplug/plug your board to do a power on reset.

retest the whole process with this branch, RTT should work flawlessly.

### Issues/PRs references
Bug found while reviewing #14416

### CODE REVIEW

File: cpu/saml1x/cpu.c [hunk 1]
```diff
@@ -135,9 +135,24 @@ void cpu_init(void)
...
+                        /* If RTC was running from a previous boot, unsure we
+                           don't disable it, otherwise the clock will be enable
+                           later if RTT or RTC is used by the application  */
+                         | ((RTC->MODE2.CTRLA.bit.ENABLE) ? MCLK_APBAMASK_RTC : 0)
 #endif
                          ;
 
+
+    /* Disable the RTC module to prevent synchronization issues during CPU init
+       if the RTC was running from a previous boot (e.g wakeup from backup)
```
Comment (fjmolinas) [inline cpu/saml1x/cpu.c:149]: Travis is reporting a white space here.

File: cpu/saml1x/cpu.c [hunk 2]
```diff
@@ -135,9 +135,24 @@ void cpu_init(void)
 #endif
 #ifdef MODULE_PERIPH_GPIO
                          | MCLK_APBAMASK_PORT
+#endif
+#ifdef MODULE_PERIPH_RTC_RTT
+                        /* If RTC was running from a previous boot, unsure we
```
Comment (benpicco) [inline cpu/saml1x/cpu.c:140]:
```suggestion
                        /* If RTC was running from a previous boot, ensure we
```

File: cpu/saml1x/cpu.c [hunk 3]
```diff
@@ -135,9 +135,24 @@ void cpu_init(void)
...
+                           don't disable it, otherwise the clock will be enable
+                           later if RTT or RTC is used by the application  */
+                         | ((RTC->MODE2.CTRLA.bit.ENABLE) ? MCLK_APBAMASK_RTC : 0)
```
Comment (benpicco) [inline cpu/saml1x/cpu.c:143]: We might as well just always enable it.

**Comment by benpicco** [comment]
I can confirm that this fixes the issue on `saml1x`.
I'm a bit puzzled why this is not an issue with the other family members too, but let's get this in to prevent nasty surprises.

**Comment by dylad** [comment]
Fixed all comments.

**Review by benpicco — APPROVED** [review]
Please squash.

**Comment by dylad** [comment]
Squashed and rebased

**Comment by dylad** [comment]
Thanks for merging !

**Pull request merged**